### PR TITLE
docs: Update PR template to demand tests.

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`. Why? for the release notes. 
* [ ] Optional. I've added My organization is added to the README.
* [ ] I've signed the CLA and required builds are green.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Checklist:
 
-* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
-* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`. Why? for the release notes. 
+* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
```
Comment (alexec) [inline .github/pull_request_template.md]: I thought this was confusing so simplified it.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
+* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.
```
Comment (alexec) [inline .github/pull_request_template.md]: Just removed the "why" from this line.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 Checklist:
 
-* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
-* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`. Why? for the release notes. 
+* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
+* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
+* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
```
Comment (alexec) [inline .github/pull_request_template.md]: Try and encourage people to write tests.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
...
 * [ ] Optional. I've added My organization is added to the README.
 * [ ] I've signed the CLA and required builds are green.
```
Comment (alexec) [inline .github/pull_request_template.md]: I've enabled "required checks".

File: .github/pull_request_template.md (hunk 3 above)
Comment (simster7) [reply .github/pull_request_template.md]: `unlike` -> `unlikely`

**Pull request merged**